src/curve.ts

```diff
@@ -18,10 +18,37 @@ import {
   curveNatural,
   curveStep,
   curveStepAfter,
-  curveStepBefore
+  curveStepBefore,
+  CurveBundleFactory,
+  CurveCardinalFactory,
+  CurveCatmullRomFactory
 } from "d3";
+import type { CurveFactory } from "d3";
```

Comment:
```suggestion
  curveStepBefore
} from "d3";
import type {CurveFactory, CurveBundleFactory, CurveCardinalFactory, CurveCatmullRomFactory} from "d3";
```

these are types from https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/d3-shape/index.d.ts

Reply:
CurveCatmullRomFactory and sisters should maybe be treated the same way as CurveFactory (in the sense that they only exist in "@types/d3", and do not belong to "d3"—at least not d3.js). I tested locally and tsc doesn't seem to choke on it—but it's possible there's something I'm not seeing.

Reply:
Ah! I see what you mean now, these are only types. Ok will update! 🙏